src/server/mod.rs

```diff
@@ -948,4 +961,35 @@ where
             },
         }
     }
+
+    fn handle_control_channel_error(error: tokio::timer::timeout::Error<FTPError>, with_metrics: bool) -> Reply {
+        return if error.is_timer() {
```

Comment:
Is there no way to pattern match this?

Reply:
Maybe I'm just doing something wrong but no, not that I could see. While the tokio::timer::timeout::Error is public, the Kind enum that it uses (Inner, Elapsed, Timer) is private.